### DISCUSSION TITLE
How can you merge two repositories while keeping all commit and contribution history?

### DISCUSSION THREAD

**Discussion by dascrapy** [community/community#75875] [Repositories] [answered]
### Select Topic Area

General

### Body

If you have two repositories and you want to unify them into one but don't want to lose your contribution or commit history.

**Comment by ghost** [comment, marked as answer]
Hello, I am not very familiar with Git, but I did manage to merge two and more repositories into one. (after a few unsuccessful attempts seeking help on the internet, I did manage to get assistance from [here](https://gist.github.com/msrose/2feacb303035d11d2d05)).

Here's what I did:
I wanted to merge `repo-a` into `repo-b`.
I opened `repo-a` with GitHub Desktop (cloned),
opened it in VSCode, and accessed the terminal in VSCode (bash).

```
$ git remote rm origin
$ git remote add origin url_to_repo-b
$ git fetch
$ git checkout main  # Adjust the branch name as needed
$ git pull origin main --allow-unrelated-histories
```

Now, all the files showed up in the same folder as `repo-b`.

```
$ git push origin main
```

I checked in GitHub Desktop if any changes were left to push and completed the process. In `repo-b`, all the files from both repositories with all commits were present. I deleted `repo-a`. In the contribution section, it is showing all commits made to `repo-b` with their respective date-time history.

Check [the repo](https://github.com/lisztomania23/frontend-mentor-challenges) I merged in.

**Reply by reuelrai** [reply]
Thank you. was having trouble getting it right but it worked perfectly for me.

**Reply by rnv812** [reply]
Thank you, my contribution history is preserved after deleting merged repos 👍🏻

**Reply by andybeet** [reply]
thanks!! worked for me too! Just need to rename repo locally (since it is now poining to other repo)

**Reply by glory0fNothing** [reply]
A bit in the future here, but I would also like to say thank you for this. I tried 3 different ways before yours and none of them worked properly. 
To anyone else doing the same, make sure that none of the files between the repos that share the same name. You'll have to change the name of one for the merge to work or it will throw an error at you.

**Comment by chandra-opsera** [comment]
git subtree add is better approach

**Comment by reuelrai** [comment]
Thanks... I will try it...

**Comment by johnmacha** [comment]
Thanks it worked for me too. Just faced some unrelated history issues but I solved it
